javascript/ql/src/Statements/UseOfReturnlessFunction.ql

```diff
@@ -0,0 +1,85 @@
+/**
+ * @name Use of returnless function
+ * @description Using the return value of a function that does not explicitly return is indicative of a mistake.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/use-of-returnless-function
+ * @tags maintainability,
+ *       correctness
+ * @precision high
+ */
+
+import javascript
+import Declarations.UnusedVariable
+import Expressions.ExprHasNoEffect
+import Statements.UselessConditional
+
+predicate returnsVoid(Function f) {
+  f.getBody() instanceof Stmt and
```

Comment:
I still think this check belongs in `returnsVoid`.   
However, I did change the check to `not f instanceof ArrowFunctionExpr`, as that better conveys the intent of the check.

Reply:
OK.
But now I may have lost track of what you are trying to do with this check then, because `  not exists(f.getAReturnedExpr())` implies `not f instanceof ArrowFunctionExpr`.

Reply:
You're right, I didn't look at the bigger picture.  
The check existed before `getAReturnedExpr` was used, and was intended to filter out the arrow-functions that returned a value. 

And my change to `not f instanceof ArrowFunctionExpr` was wrong, as an arrow-function can contain a block-statement. 

You're right that `not exists(f.getAReturnedExpr())` makes the check redundant, so I removed it.